src/main/java/com/aws/greengrass/mqtt/bridge/clients/LocalMqtt5Client.java

```diff
@@ -687,26 +703,57 @@ private Mqtt5Client createCrtClient() throws MessageClientException {
     }
 
     void setClient(Mqtt5Client client) {
-        if (isSSL()) {
-            mqttClientKeyStore.listenToCAUpdates(onKeyStoreUpdate);
-        }
         this.client = client;
     }
 
     private boolean isSSL() {
         return "ssl".equalsIgnoreCase(brokerUri.getScheme());
     }
 
-    void reset() { // TODO callback shouldn't be synchronous
-        stop();
-        try {
-            setClient(createCrtClient());
-        } catch (MessageClientException e) {
-            // TODO recover
-            LOGGER.atError().cause(e).log("unable to start mqtt client during reset");
-            return;
+    /**
+     * Stop and start the mqtt client.
+     */
+    public void reset() {
+        synchronized (resetLock) {
+            // we may reset through client cert or ca change, cancel any pending reset attempts
+            cancelResetRetry();
+            closeClient();
+
+            try {
+                this.client = clientSupplier.apply();
+            } catch (MessageClientException e) {
+                LOGGER.atWarn().cause(e).log("Unable to create mqtt client, will retry");
+                scheduleResetRetry();
+                return;
+            }
+
+            try {
+                client.start();
+            } catch (CrtRuntimeException e) {
+                LOGGER.atWarn().cause(e).log("Unable to start mqtt client, will retry");
+                closeClient();
+                scheduleResetRetry();
+            }
+        }
+    }
+
+    private void scheduleResetRetry() {
+        synchronized (resetLock) {
+            cancelResetRetry();
+            resetRetryTask = ses.schedule(this::reset, resetDelay.getSeconds(), TimeUnit.SECONDS);
+            resetDelay = resetDelay.plus(resetDelay); // exponential backoff
```

Comment:
resetDelay needs to be set back to default on success

Reply:
fixed